src/libNetwork/P2PComm.cpp

```diff
@@ -204,22 +209,93 @@ bool SendJob::SendMessageSocketCore(const Peer& peer, const bytes& message,
     serv_addr.sin_addr.s_addr = peer.m_ipAddress.convert_to<unsigned long>();
     serv_addr.sin_port = htons(peer.m_listenPortHost);
 
-    if (connect(cli_sock, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) <
-        0) {
-      LOG_GENERAL(WARNING, "Socket connect failed. Code = "
-                               << errno << " Desc: " << std::strerror(errno)
-                               << ". IP address: " << peer);
-      if (P2PComm::IsHostHavingNetworkIssue()) {
-        LOG_GENERAL(WARNING, "[blacklist] Encountered "
-                                 << errno << " (" << std::strerror(errno)
-                                 << "). Adding " << peer.GetPrintableIPAddress()
-                                 << " to blacklist");
-        Blacklist::GetInstance().Add(peer.m_ipAddress);
+    // Set non-blocking
+    if ((arg = fcntl(cli_sock, F_GETFL, NULL)) < 0) {
+      LOG_GENERAL(WARNING, "couldn't get flags on socket");
+      return false;
+    }
+    arg |= O_NONBLOCK;
+    if (fcntl(cli_sock, F_SETFL, arg) < 0) {
+      LOG_GENERAL(WARNING, "couldn't set socket as non-blocking");
+      return false;
+    }
+
+    int status;
+    bool connectStat = true;
+    if ((status = connect(cli_sock, (struct sockaddr*)&serv_addr,
```

Comment:
It looks like if `connect()` fails then `connectStat` remains `true`.  Is the flow for that case OK?